packages/connect-core/src/utils/descriptor/describe.ts

```diff
@@ -0,0 +1,116 @@
+import { providers as ethersProviders } from 'ethers'
+
+import {
+  tryEvaluatingRadspec,
+  tryDescribingUpdateAppIntent,
+  postprocessRadspecDescription,
+} from '../radspec/index'
+import { StepDecoded, StepDescribed, PostProcessDescription } from '../../types'
+import App from '../../entities/App'
+import Transaction from '../../entities/Transaction'
+
+export async function describeStep(
+  step: StepDecoded,
+  installedApps: App[],
+  provider: ethersProviders.Provider
+): Promise<StepDescribed> {
+  let decoratedStep
+  // TODO: Add once we support intent Basket
+  // if (Array.isArray(step)) {
+  //   // Intent basket with multiple transactions in a single callscript
+  //   // First see if the step can be handled with a specialized descriptor
+  //   try {
+  //     decoratedStep = await tryDescribingUpgradeOrganizationBasket(step)
+  //   } catch (err) {}
+
+  //   // If the step wasn't handled, just individually describe each of the transactions
+  //   return decoratedStep || describePath(step, installedApps, provider)
+  // }
+
+  // Single transaction step
+  // First see if the step can be handled with a specialized descriptor
+  try {
+    decoratedStep = await tryDescribingUpdateAppIntent(step, installedApps)
+  } catch (err) {}
+
+  // Finally, if the step wasn't handled yet, evaluate via radspec normally
+  if (!decoratedStep) {
+    try {
+      decoratedStep = await tryEvaluatingRadspec(step, installedApps, provider)
```

Comment:
We try to evaluate using Radspec. Here is where we would like further flexibility to the Connect use and provider more Radspec flags that allow for specialized descriptions for the particular organization.